src/ctapipe/irf/optimize.py

```diff
@@ -129,7 +136,16 @@ def read(cls, file_name):
             gh_cuts = QTable.read(hdul[2])
             valid_energy = QTable.read(hdul[3])
             valid_offset = QTable.read(hdul[4])
-            spatial_selection_table = QTable.read(hdul[5]) if len(hdul) > 5 else None
+            spatial_selection_table = None
+            multiplicity_cuts = None
+            if len(hdul) == 6:
+                if hdul[5].header["EXTNAME"] == "RAD_MAX":
+                    spatial_selection_table = QTable.read(hdul[5])
```

Comment:
you can just index by extname `hdul["RAD_MAX"]` and also check `"RAD_MAX" in hdul`

Reply:
do not rely on indicies